src/lucky/secure_headers/set_frame_guard.cr

```diff
@@ -0,0 +1,49 @@
+module Lucky
+  module SecureHeaders
+    # This module sets the HTTP header [X-Frame-Options](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/X-Frame-Options).
+    # It's job is responsible for deciding which site can call your site from within a frame.
+    # For more information, read up on [Clickjacking](https://en.wikipedia.org/wiki/Clickjacking).
+    #
+    # Include this module in the actions you want to add this to.
+    # A required method `frame_guard_options` must be defined`
+    # ```
+    # class BrowserAction < Lucky::Action
+    #   include Lucky::SecureHeaders::SetFrameGuard
+    #
+    #   def frame_guard_options
+    #     {allow_from: "deny"}
+    #   end
+    # end
+    # ```
+    #
+    # ### Options
+    # The `frame_guard_options` method must be defined and return `NamedTuple(allow_from: String)`
+    # The `allow_from` key can have one of 3 String values:
+    # - `"same"` or `"sameorigin"`
+    # - `"nowhere" or `"deny"`
+    # - a valid URL e.g. `"https://mysite.com"`
+    module SetFrameGuard
+      macro included
+        before set_frame_guard_header
+      end
+
+      abstract def frame_guard_options : NamedTuple(allow_from: String)
+
+      private def set_frame_guard_header
+        context.response.headers["X-Frame-Options"] = frame_guard_value(frame_guard_options)
+        continue
+      end
+
+      private def frame_guard_value(options)
+        case options[:allow_from].downcase
+        when "same", "sameorigin"
```

Comment:
How do you feel about removing this case statement and just passing the value through? Is there a danger to setting `same` or `nowhere`? Or maybe there are conventions I'm missing. I could also see going the other way of throwing an error if the values are anything but `sameorigin`, `deny`, or a valid URL. What do you think?

Reply:
Well, there's only 3 possible options as far as browsers go, so it has to be `sameorigin`, `deny`, or a valid URL. I liked the "nowhere" that helmet is using because it reads nice `allow_from: "nowhere"`; however, that's not necessary, and does add a little extra cruft. I'm down for just passing in the value and raising an exception when it's not one of those 3. If you're cool with that, I'll make that change

Reply:
I love the idea of raising if it doesn't match those 3!

Reply:
Ok, made that change. If y'all are cool with that, then I'll implement the other two 😄